pandas/core/indexes/datetimelike.py

```diff
@@ -809,8 +809,9 @@ def take(
             self, indices, axis, allow_fill, fill_value, **kwargs
         )
 
-        maybe_slice = lib.maybe_indices_to_slice(indices, len(self))
-        if isinstance(maybe_slice, slice):
-            freq = self._data._get_getitem_freq(maybe_slice)
-            result._data._freq = freq
+        if indices.ndim == 1:
```

Comment:
this seems very weird.  if indices.ndim > 1 then result.ndim > 1 and that cant be valid for an Index?

Reply:
Indeed, I think I need to find another way to approach this. The following is for Int64, I figure someone similar applies to datetimelike (but haven't checked).

```
index = pd.Index([1, 2, 3], dtype="Int64")
indices = np.array([[0, 1], [-1, 2]])
result = index.take(indices)

print(result)
# Index([[1, 2], [3, 3]], dtype='Int64')

print(result.values)
# <IntegerArray>
# [
# [1, 2],
# [3, 3]
# ]
# Shape: (2, 2), dtype: Int64
```